assets/js/blocks/product-category/block.js

```diff
@@ -37,8 +37,13 @@ class ProductByCategoryBlock extends Component {
 		this.state = {
 			products: [],
 			loaded: false,
+			changedAttributes: null,
+			isEditing: false,
 		};
-
+		this.startEditing = this.startEditing.bind( this );
+		this.stopEditing = this.stopEditing.bind( this );
+		this.setAttributes = this.setAttributes.bind( this );
+		this.save = this.save.bind( this );
```

Comment:
Correct. :+1: 

This is usually needed when passing the function as a callback (for example, when assigning the function to an `onClick` or similar attributes in a React component), because it loses the context and `this` would be `undefined`.